Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -3748,6 +3748,47 @@ JS_DEFINE_NATIVE_FUNCTION(@class_name@::@attribute.getter_callback@)
     if (content_attribute_value.has_value())
         retval = MUST(Infra::convert_to_scalar_value_string(*content_attribute_value));
 )~~~");
+            }
+            // If a reflected IDL attribute has the type T?, where T is either Element
+            // FIXME: or an interface that inherits from Element,
+            // then with attr being the reflected content attribute name:
+            else if (attribute.type->is_nullable() && attribute.type->name() == "Element") {
+                // The getter steps are to return the result of running this's get the attr-associated element.
+                attribute_generator.append(R"~~~(
+    auto retval = GC::Ptr<Element> {};                
+)~~~");
+
+                // 1. Let element be the result of running reflectedTarget's get the element.
+                // 2. Let contentAttributeValue be the result of running reflectedTarget's get the content attribute.
+                attribute_generator.append(R"~~~(
+    auto contentAttributeValue = impl->attribute(HTML::AttributeNames::@attribute.reflect_name@);
+)~~~");
+                // 3. If reflectedTarget's explicitly set attr-element is not null:
+                //      1. If reflectedTarget's explicitly set attr-element is a descendant of any of element's shadow-including ancestors, then return reflectedTarget's explicitly set attr-element.
+                //      2. Return null.
+                attribute_generator.append(R"~~~(
+    auto const explicitly_set_attr = TRY(throw_dom_exception_if_needed(vm, [&] { return impl->get_@attribute.cpp_name@(); }));
+    if (explicitly_set_attr) {
+        if (impl->shadow_including_root().is_equal_node(&explicitly_set_attr->shadow_including_root())) {
+            retval = explicitly_set_attr;
+        }
+        retval = explicitly_set_attr;
+    }
+)~~~");
+                // 4. Otherwise, if contentAttributeValue is not null, return the first element candidate, in tree order, that meets the following criteria:
+                //      candidate's root is the same as element's root;
+                //      candidate's ID is contentAttributeValue; and
+                //      candidate implements T.
+                // If no such element exists, then return null.
+                // 5. Return null.
+
+                // FIXME: This works when T is Element but might need adjustment when we handle subtypes too
+                attribute_generator.append(R"~~~(
+    else if (contentAttributeValue.has_value()) {
+        retval = impl->document().get_element_by_id(contentAttributeValue.value());
```

Comment:
Using `get_element_by_id()` here could return an element that doesn't share element's root.

Instead you could use `for_each_in_inclusive_subtree_of_type`:
```c++
        impl->root().for_each_in_inclusive_subtree_of_type<DOM::Element>([&](auto& candidate) {
            if (candidate.attribute(HTML::AttributeNames::id) == contentAttributeValue.value()) {
                retval = &candidate;
                return TraversalDecision::Break;
            }
            return TraversalDecision::Continue;
        });
```